### PR TITLE
Add battery entries to plexos_properties

### PR DESCRIPTION
**Pull request by MartySchwarz**

### CODE REVIEW

**Comment by MartySchwarz** [comment]
Subtract batterie_load from ExtraPropertiesPLEXOS.demand() before merging.

File: marmot/plottingmodules/generation_stack.py [hunk 1]
```diff
@@ -413,13 +407,34 @@ def gen_stack(
...
+                ].get(scenario)
+
+                if stacked_bat_gen.empty is True:
+                    logger.info("No Battery generation in selected Date Range")
+                else:
+                    if shift_leapday:
+                        stacked_bat_gen = self.adjust_for_leapday(stacked_bat_gen)
+
+                    stacked_bat_gen = stacked_bat_gen.xs(
+                        'GVEA BESS', level='battery_name'
```
Comment (daniellevie) [inline marmot/plottingmodules/generation_stack.py:422]: 'GVEA BESS' may cause issues if this battery_name is not in your results. What exactly are you trying to do here?

**Review by daniellevie — COMMENTED** [review]
@MartySchwarz we should probably have a call to quickly review before merging. 

Some pieces of code I'm unsure of

**Comment by MartySchwarz** [comment]
@Levie, ***@***.***>

Happy to hop on a call. We can remove that line that references GVEA BESS, it was a quick fix to plot something right before a meeting.

From: Daniel Levie ***@***.***>
Date: Monday, October 3, 2022 at 12:52 PM
To: NREL/Marmot ***@***.***>
Cc: Schwarz, Marty ***@***.***>, Mention ***@***.***>
Subject: Re: [NREL/Marmot] Add battery entries to plexos_properties (PR #46)

CAUTION: This email originated from outside of NREL. Do not click links or open attachments unless you recognize the sender and know the content is safe.

@daniellevie commented on this pull request.

@MartySchwarz<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartySchwarz&data=05%7C01%7CMarty.Schwarz%40nrel.gov%7Cbb363241824a4598176508daa5707331%7Ca0f29d7e28cd4f5484427885aee7c080%7C0%7C0%7C638004199656198318%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2FcKMSnRktur%2BXOXSNbY4OZL76PJu3mBr9xGV78%2BFiMg%3D&reserved=0> we should probably have a call to quickly review before merging.

Some pieces of code I'm unsure of

________________________________

In marmot/plottingmodules/generation_stack.py<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNREL%2FMarmot%2Fpull%2F46%23discussion_r986116892&data=05%7C01%7CMarty.Schwarz%40nrel.gov%7Cbb363241824a4598176508daa5707331%7Ca0f29d7e28cd4f5484427885aee7c080%7C0%7C0%7C638004199656198318%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=fZwgke4SxKJwvQKSBFGdkAdNGu8%2FRJ7K4AJgeXtm0Pc%3D&reserved=0>:

> @@ -413,13 +407,34 @@ def gen_stack(

                 else:

                     vre_gen_cat = self.gen_categories.vre



+                #Insert battery generation.

+                stacked_bat_gen : pd.DataFrame = self[

+                    f"batterie_Generation{data_resolution}"

+                ].get(scenario)

+

+                if stacked_bat_gen.empty is True:

+                    logger.info("No Battery generation in selected Date Range")

+                else:

+                    if shift_leapday:

+                        stacked_bat_gen = self.adjust_for_leapday(stacked_bat_gen)

+

+                    stacked_bat_gen = stacked_bat_gen.xs(

+                        'GVEA BESS', level='battery_name'

'GVEA BESS' may cause issues if this battery_name is not in your results. What exactly are you trying to do here?

**Comment by MartySchwarz** [comment]
Incorporate battery load to load/demand properties in extraproperties.py in the formatter step, rather than subtracting them from load in the plotting step.

**Comment by daniellevie** [comment]
@MartySchwarz is this PR needed anymore? Or should we close and delete it?

**Comment by MartySchwarz** [comment]
These changes were made obselete by https://github.com/NREL/Marmot/pull/49

**Pull request closed without merging**